### PR TITLE
print cause when there is one

### LINKED ISSUE #33: Error() omits the cause of error
Unlike `errors.Wrap`, the `(*Error).Wrap` of RFC errors makes an error with a `Error()` method that wipes all the information the `cause` error has.

It would be good if we can reserve them.

**Possible risk:** some tests might fail?

### PR DESCRIPTION
**Pull request by YuJuncen**
Fix #33

### CODE REVIEW

**Comment by kennytm** [comment]
Please add a unit test. 

Rest lgtm.

**Comment by YuJuncen** [comment]
@kennytm Done

File: normalize_test.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	"regexp"
+	"testing"
+)
+
+func errorMatches(t *testing.T, err error, re string) {
+	if err == nil && re != "" {
+		t.Errorf("nil error doesn't match %s", re)
+		return
+	}
+	match, reErr := regexp.MatchString(re, err.Error())
```
Comment (kennytm) [inline normalize_test.go:13]: why use a regexp when the test only checks for literal strings 🤔.

Comment (YuJuncen) [reply normalize_test.go:13]: At line 32, regexp was used for check whether the error string was exactly the pattern.

```
errorMatches(t, notWrapped, `^\[Internal:Test\]everything is alright$`)
```

Comment (kennytm) [reply normalize_test.go:13]: @YuJuncen  isn't line 29 also checking for exact match 🤔

Comment (YuJuncen) [reply normalize_test.go:13]: > MatchString reports whether the string s contains any match of the regular expression pattern. More complicated queries need to use Compile and the full Regexp interface.

@kennytm It matches any substring.

Comment (YuJuncen) [reply normalize_test.go:13]: Anyway, this might be useful in the future... (I hope...)

**Review by lysu — APPROVED** [review]
LGTM

**Pull request merged**